File: 9-data_structures_stacks+queues/stack_lst.py
class Stack:
    def __init__(self):
        self.lst = []

    def peek(self):
        if len(self.lst) == 0:
            return None

        return self.lst[-1]

    def push(self, value):
        self.lst.append(value)

    def pop(self):
        if len(self.lst) == 0:
            return None

        self.lst.pop()


s = Stack()
s.push("google")
s.push("udemy")
s.push("discord")
print(s.peek())
s.pop()
s.pop()
s.pop()
print(vars(s))

File: 9-data_structures_stacks+queues/stacks/lib/stack_lst.ex
defmodule StackLst do
  defstruct lst: []
  alias StackLst, as: S

  def new(), do: %S{}

  def peek(%S{lst: []}), do: nil
  def peek(%S{lst: [h | _]}), do: h

  def push(%S{lst: lst}, value), do: %S{lst: [value | lst]}

  def pop(%S{lst: []}), do: nil
  def pop(%S{lst: [_ | t]}), do: %S{lst: t}
end

# StackLst.new()
# |> StackLst.push('google')
# |> StackLst.push('udemy')
# |> StackLst.push('discord')
